### ISSUE TITLE
Access to the path 'Microsoft.ApplicationInsights.Agent.Telemetry.dll' is denied.

### ISSUE THREAD

**Issue by opencloudcode** [microsoft/ApplicationInsights-Home#241]
Issue with trying to do Update-ApplicationInsightsVersion

```
PS C:\Users\userr>         Update-ApplicationInsightsVersion;
Update-ApplicationInsightsVersion : Access to the path 'Microsoft.ApplicationInsights.Agent.Telemetry.dll' is denied.
```

I can update by opening Microsoft.Diagnostics.Agent.StatusMonitor.Helper and updating it manually from there. Is there a way to script  that process as a tempary fix?

Can you guys help asap, this issue occuring in production releases

**Comment by SergeyKanzhelev** [comment]
Are you running script under admin console? This maybe simply a permission issue

**Comment by opencloudcode** [comment]
@SergeyKanzhelev  running as admin

**Comment by SergeyKanzhelev** [comment]
Do you have this file under `C:\Program Files\Microsoft Application Insights\Status Monitor\Bin\<version>`? One workaround I can think of is to try to copy all files from here and `C:\Program Files\Microsoft Application Insights\Status Monitor\PowerShell` into a single folder and see if this will help.

**Comment by SergeyKanzhelev** [comment]
CC: @Mikhail-Pranovich, @AlexKlimovMS and @iusafaro

**Comment by Mikhail-Pranovich** [comment]
Could you check what process locks that file? (the sysinternalsuite/process explorer can be used for that)

**Comment by Blackbaud-ChrisBlythe** [comment]
A search for the dll only shows the current powershell process where I imported the App Insights module...

![image](https://user-images.githubusercontent.com/14931768/45648387-0d88ed80-ba8e-11e8-9054-288ae6883830.png)

Here is the full txt of the exception:
`PS C:\Users\cjblythe> Update-ApplicationInsightsVersion
Update-ApplicationInsightsVersion : Access to the path 'Microsoft.ApplicationInsights.Agent.Telemetry.dll' is denied.
At line:1 char:1
+ Update-ApplicationInsightsVersion
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Update-ApplicationInsightsVersion], Exception
    + FullyQualifiedErrorId : System.UnauthorizedAccessException`

**Comment by Blackbaud-ChrisBlythe** [comment]
Is there a way to get any traction here? In general, it seems like the Status Monitor has gotten pretty stale. Even in UI mode, it seems like it can only upgrade to SDK version 2.4.0 while the latest available version is now 2.8.1. 

@SergeyKanzhelev Are there any updates coming to this tool to address the growing number of issues?

**Comment by MS-TimothyMothra** [comment]
Hello All,
I’ve just published v0.1-alpha of IISConfigurator. This is our replacement for StatusMonitor.

I’ve organized all our docs for the IISConfigurator here:  https://github.com/Microsoft/ApplicationInsights-Home/blob/master/Samples/IISConfigurator/TOC.md
This includes:
	- A Quick Start Guide
	- Detailed getting started instructions
	- API reference for the new PowerShell Module
	- A page dedicated to Troubleshooting and Known Issues


Disclaimer: This is a prototype application. We do not recommend using this on your production environments.
Please pay close attention to the known issues.
Your apps may have issues if they include any of the following DLLs in the bin directory:
	· Microsoft.ApplicationInsights.dll
	· Microsoft.AspNet.TelemetryCorrelation.dll
	· System.Diagnostics.DiagnosticSource.dll
 


I am very interested in your feedback!
Please don’t hesitate to contact me if you have any issues or comments.
Specifically,
	- I would love to hear about your experiences getting started
	- And what you would like to see moving forward

Thank you,

**Comment by MS-TimothyMothra** [comment]
Today I've released the Release Candidate for Status Monitor v2.
Because this is the replacement for the original Status Monitor, I'm going to close this issue.

Please review or documentation to get started: https://docs.microsoft.com/en-us/azure/azure-monitor/app/status-monitor-v2-overview
